coreneuron/nrniv/output_spikes.cpp

```diff
@@ -95,8 +95,12 @@ void local_spikevec_sort(std::vector<double>& isvect,
 #if NRNMPI
 
 void sort_spikes(std::vector<double>& spikevec_time, std::vector<int>& spikevec_gid) {
-    double lmin_time = *(std::min_element(spikevec_time.begin(), spikevec_time.end()));
```

Comment:
good catch. Maybe it would be good to also guard against the case where there were no spikes at all. maybe with something like
```c++
auto nspikes = nrnmpi_int_allmax(spikevec_time.size());
if (nspikes == 0) {
    return;
}
```

Reply:
I totally agree with this conceptually, however everything should work fine even with 0 total spikes.
Do you think it worth the time spent doing the allreduce on the 99.9% of cases where there would be spikes against the time saved when there are 0 spikes?

Reply:
You are right of course, this is overkill and is implicitly handled further down.